packages/components/addon/components/hds/code-block/index.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { htmlSafe } from '@ember/template';
+import { guidFor } from '@ember/object/internals';
+import Prism from 'prismjs';
+
+import 'prismjs/plugins/line-numbers/prism-line-numbers';
+import 'prismjs/plugins/line-highlight/prism-line-highlight';
+
+import 'prismjs/components/prism-bash';
+import 'prismjs/components/prism-go';
+import 'prismjs/components/prism-hcl';
+import 'prismjs/components/prism-json';
+import 'prismjs/components/prism-log';
+import 'prismjs/components/prism-shell-session';
+import 'prismjs/components/prism-yaml';
```

Comment:
What if other prism plugins are loaded elsewhere? Does Prism keep track of plugins in a global shared var?

If a user wanted to add a plugin to add some functionality, could a hook be added here for that?

I ask because I'm looking at a filtering interface right now where you there's a JSON CodeBlock - it'd be amazing to be able to click on the tokens or lines and update the search params for the interface based on that.

That's not something I'd expect the DS CodeBlock to support, but if there was a way to tell the component to "run this plugin on the prisma bits before displaying it", it would mean I wouldn't have to start from scratch.

Reply:
Hey @meirish – I wanted to acknowledge your comment and try to answer your questions.
Unfortunately Prism is indeed stored in a global variable and we failed to create an alias for it (such as HdsPrism) because parts of the code (and the plugins) look specifically for a `Prism` global to exist (I will not comment on this choice). About your suggestion for a hook, I made a note to look into it in the future.